src/handle-data.js

```diff
@@ -29,24 +35,30 @@ export function getColumnInfo(layout, colIndex) {
 }
 
 export default async function manageData(model, layout, pageInfo, setPageInfo) {
-  const { page, rowsPerPage } = pageInfo;
-  const top = page * rowsPerPage;
+  const { page, rowsPerPage, rowsPerPageOptions } = pageInfo;
   const size = layout.qHyperCube.qSize;
-  const allRowsLength = size.qcy;
+  const top = page * rowsPerPage;
+  const width = size.qcx;
+  const totalHeight = size.qcy;
+  const height = Math.min(rowsPerPage, totalHeight - top);
```

Comment:
based on https://qlik.dev/apis/json-rpc/qix/genericobject#%23%2Fentries%2FGenericObject%2Fentries%2FGetHyperCubeData,

how about 
const qHyperCube = layout.qHyperCube;
const size = qHyperCube.qiszel
const top = page * rowsPerPage;
const width = size.qcx;
const height = size.qcy;
const actualHeight = Math.min(rowsPerPage, totalHeight - top);

Reply:
I don't see how the docs makes your suggestion make more sense. My thought was that width and height is what will be used for the page.

Reply:
<img width="467" alt="Screenshot 2021-11-30 at 09 41 43" src="https://user-images.githubusercontent.com/4471489/144013845-b4859b77-3224-4c9a-aa47-eaec7e39b3ef.png">
I thought that width and height are what we got from the engine and make it clear that actualHeight is what is used for the page. now I think that both are fine.